src/core/python2js.c

```diff
@@ -557,45 +525,135 @@ EM_JS_REF(JsRef, _JsArray_PostProcess_helper, (JsRef jscontext, JsRef array), {
     Hiwire.get_value(jscontext).dict_converter(Hiwire.get_value(array)));
 })
 
+// clang-format off
+EM_JS_REF(
+JsRef,
+python2js__default_converter,
+(JsRef jscontext, PyObject* object),
+{
+  let context = Hiwire.get_value(jscontext);
+  let proxy = Module.pyproxy_new(object);
+  let result = context.default_converter(
+    proxy,
+    context.converter,
+    context.cacheConversion
+  );
+  proxy.destroy();
+  return Hiwire.new_value(result);
+})
+// clang-format on
+
 static JsRef
 _JsArray_PostProcess(ConversionContext context, JsRef array)
 {
   return _JsArray_PostProcess_helper(context.jscontext, array);
 }
 
+// clang-format off
+EM_JS_REF(
+JsRef,
+python2js_custom__create_jscontext,
+(ConversionContext context,
+  PyObject* cache,
+  JsRef dict_converter,
+  JsRef default_converter),
+{
+  let jscontext = {};
+  if (dict_converter !== 0) {
+    jscontext.dict_converter = Hiwire.get_value(dict_converter);
+  }
+  if (default_converter !== 0) {
+    jscontext.default_converter = Hiwire.get_value(default_converter);
+    jscontext.cacheConversion = function (input, output) {
+      // input should be a PyProxy, output should be a Javascript
+      // object
+      if (!API.isPyProxy(input)) {
+        throw new Error("Oops!");
```

Comment:
A little more concrete error message would be great.

Reply:
Oh yeah I meant to go back and fix this 😂